ns for fx: add fqn to results, when present

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61380
* **#61377**
* #61360
* #61359
* #61358
* #61357
* #61323
* #61129

Summary:

Both the quantization tracer and the NS tracer record
`_node_name_to_scope`, which contains the mapping from
node name to FQN.

This PR adds the FQN information to the NS results, so that it is
more convenient for users to attribute a NS result to the corresponding
module in their model.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_match_activations_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_shadow_activations_fqn
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29600349](https://our.internmc.facebook.com/intern/diff/D29600349)

vkuzo committed:
ns for fx: add fqn to results, when present

Summary:

Both the quantization tracer and the NS tracer record
`_node_name_to_scope`, which contains the mapping from
node name to FQN.

This PR adds the FQN information to the NS results, so that it is
more convenient for users to attribute a NS result to the corresponding
module in their model.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_match_activations_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_shadow_activations_fqn
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29600349).

hx89 commented on torch/quantization/ns/utils.py:
```diff
@@ -354,6 +354,42 @@ def rekey_logger_info_on_node_name_of_model(
+
+    A common use case benefitting from this is comparing a model prepared by
+    quantization to a quantized model. In this case, the model prepared by
+    quantization would have `fqn` entries, and the quantized model would not.
```
Wondering why quantized model wouldn't have 'fqn' entries?

jerryzh168 replied:
I think maybe we can add node_name_to_scope to quantized model as well if that's needed, we just don't have a clear use case yet

jerryzh168 commented on torch/quantization/ns/utils.py:
```diff
@@ -354,6 +354,42 @@ def rekey_logger_info_on_node_name_of_model(
+                    if model_results[0]['fqn'] is not None:
+                        model_name_with_fqns = model_name
+                        break
+            break
```
nit: can we just get the first result instead of using these breaks..

jerryzh168 reviewed (approved):
LG!

vkuzo committed:
Update on "ns for fx: add fqn to results, when present"


Summary:

Both the quantization tracer and the NS tracer record
`_node_name_to_scope`, which contains the mapping from
node name to FQN.

This PR adds the FQN information to the NS results, so that it is
more convenient for users to attribute a NS result to the corresponding
module in their model.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_match_activations_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_shadow_activations_fqn
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29600349](https://our.internmc.facebook.com/intern/diff/D29600349)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29600349).

vkuzo committed:
Update on "ns for fx: add fqn to results, when present"


Summary:

Both the quantization tracer and the NS tracer record
`_node_name_to_scope`, which contains the mapping from
node name to FQN.

This PR adds the FQN information to the NS results, so that it is
more convenient for users to attribute a NS result to the corresponding
module in their model.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_match_activations_fqn
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_shadow_activations_fqn
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D29600349](https://our.internmc.facebook.com/intern/diff/D29600349)

[ghstack-poisoned]

vkuzo commented:
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29600349).
